cynepco3hahue/log-extractor
# RFC: Refactor log-extractor to be modular, generic regarding input files and testable.

This patch introduces the following changes:

CORE:

  * Add the ability to parse logs from various sources. (url, zip, directory)
  * Introduce single parameter for input called source. Remove all other related params.
  * Refactor script to not unpack all files to utilize better disk space.

TESTING:

  * Create test folder to prepare ground for future tests.
  * Add simple tests for LZMAFile class.
  * Add tox.ini file.
  * Add .travis.yaml file

DEVELOPER:

  * Rename log_extractor.py to extractor.py: module name should not be same
	as package name if we want to import from it.
  * Change setup.py to use pbr [1]
  * Separate requirements file into requuirements.txt to help with reusability.

[1]: https://docs.openstack.org/pbr/latest/
